incubator/hnc/pkg/reconcilers/object_test.go

```diff
@@ -118,6 +118,8 @@ var _ = Describe("Secret", func() {
 	It("should overwrite the propagated ones if the source is updated", func() {
 		setParent(ctx, barName, fooName)
 		setParent(ctx, bazName, barName)
+		// Wait 1 second to make sure the source get propagated.
+		time.Sleep(1 * time.Second)
 		Eventually(isModified(ctx, fooName, "foo-role")).Should(BeFalse())
```

Comment:
Why is `Eventually` not enough here? Why does this test (but only this test) need an additional sleep?

Reply:
Looks like `isModified` is nore turning `func() bool`, instead it returns `bool`. IIUC, `Eventually` will not retry in this case ([details](https://moddoc.marwan.io/github.com/onsi/gomega/@v/v1.4.2#Eventually)).

> If Eventually is passed an actual that is a function taking no arguments and returning at least one value, then Eventually will call the function periodically and try the matcher against the function's first return value.

Try wrap it with `func() bool` to see if it works.

Reply:
Thanks for the comments! The `isModified` is changed to return `func() bool` in #508

Reply:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_multi-tenancy/502/pull-hnc-test/1236025480545570816

This is the error mentioned in the commit message.

Reply:
For the record (since that log file will eventually be deleted):

```
• Failure [0.079 seconds]
Secret
/home/prow/go/src/sigs.k8s.io/multi-tenancy/incubator/hnc/pkg/reconcilers/object_test.go:16
  should overwrite the propagated ones if the source is updated [It]
  /home/prow/go/src/sigs.k8s.io/multi-tenancy/incubator/hnc/pkg/reconcilers/object_test.go:118
  Expected success, but got an error:
      
      <*errors.StatusError | 0xc0002f2ab0>: {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {SelfLink: "", ResourceVersion: "", Continue: ""},
              Status: "Failure",
              Message: "Role.rbac.authorization.k8s.io \"foo-role\" not found",
              Reason: "NotFound",
              Details: {
                  Name: "foo-role",
                  Group: "rbac.authorization.k8s.io",
                  Kind: "Role",
                  UID: "",
                  Causes: nil,
                  RetryAfterSeconds: 0,
              },
              Code: 404,
          },
      }
      Role.rbac.authorization.k8s.io "foo-role" not found
  /home/prow/go/src/sigs.k8s.io/multi-tenancy/incubator/hnc/pkg/reconcilers/object_test.go:125
```